conbench/entities/benchmark_result.py

```diff
@@ -34,8 +35,12 @@ class BenchmarkResult(Base, EntityMixin):
     info_id = NotNull(s.String(50), s.ForeignKey("info.id"))
     context_id = NotNull(s.String(50), s.ForeignKey("context.id"))
     run_id = NotNull(s.Text, s.ForeignKey("run.id"))
-    data = Nullable(postgresql.ARRAY(s.Numeric), default=[])
-    times = Nullable(postgresql.ARRAY(s.Numeric), default=[])
+
+    # These can be emtpy lists. An item in the list is of type float, which
+    # includes `math.nan` as valid value.
+    data: list[float] = Nullable(postgresql.ARRAY(s.Numeric), default=[])
+    times: list[float] = Nullable(postgresql.ARRAY(s.Numeric), default=[])
```

Comment:
Important decisions in here. Maybe not all of them are good / correct.

The current type annotation should maybe be `Optional[list[float]]` because the Column is marked as `Nullable`. But I wonder: why would we want to store null/None _instead of_ an emtpy list? What's the semantic difference between both situations? Maybe we only need the empty list. In that case we remove the `Nullable`, and the type is correct.

A helpful quote from the SQLAlchemy docs:

> [Numeric](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Numeric) returns Python decimal.Decimal objects by default, based on the default value of True for the [Numeric.asdecimal](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Numeric.params.asdecimal) parameter. If this parameter is set to False, returned values are coerced to Python float objects.

Btw, I am pretty sure that `s.Numeric` the way we use it here allows for saving `NaN` as the individual array item. However, I did not quickly find confirmation for that in the SQLAlchemy docs.

Behavior that matters:
```
>>> Decimal(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: conversion from NoneType to Decimal is not supported
>>> Decimal('NaN')
Decimal('NaN')
```

Reply:
If I'm following correctly (this is about if the list can have _any_ `null` elements, yeah? Like `[0.01, null, 0.015]`? Or only cases like `[null]`?), this is ultimately to support folks uploading results even if one (or more) iterations failed. 

The docs for that are at https://github.com/conbench/conbench/blob/aa95b6d52ff6cb6697b7ab8d04c43d6cf3612860/conbench/entities/benchmark_result.py#L239 (these could and must be surface better than where they are currently, but it's what we've got given our current swagger setup).

Reply:
> (this is about if the list can have any null elements, yeah? Like [0.01, null, 0.015]? Or only cases like [null]?),

no it's really about 

> why would we want to store null/None instead of an emtpy list? 

because the database schema as-is allows for storing null instead of a list/array. I don't think this should be allowed.

> What's the semantic difference between both situations? 

(`null` vs. `[]`).